pkg/tfbridge/detailed_diff.go

```diff
@@ -318,27 +377,15 @@ func (differ detailedDiffer) makeListDiff(
 		}
 	}
 
-	simplerDiff, isSimplified := differ.simplifyDiff(diff, path, old, new)
-	if isSimplified {
-		return simplerDiff
-	}
-
 	return diff
 }
 
 func (differ detailedDiffer) makeMapDiff(
 	path propertyPath, old, new resource.PropertyValue,
 ) map[detailedDiffKey]*pulumirpc.PropertyDiff {
+	oldMap := old.ObjectValue()
+	newMap := new.ObjectValue()
```

Comment:
Just so I understand, this is ok because of `isTypeShapeMismatched`?

Reply:
yeah, I moved all the asserts there